### PR TITLE
IO fixes

### PR DESCRIPTION
**Pull request by vtjnash**
just going back through old IO issues and trying to close a few

### CODE REVIEW

File: base/REPL.jl [hunk 1]
```diff
@@ -163,6 +163,7 @@ function run_repl(repl::AbstractREPL, consumer = x->nothing)
     repl_channel = Channel(1)
     response_channel = Channel(1)
     backend = start_repl_backend(repl_channel, response_channel)
+    consumer(backend)
```
Comment (tkelman) [inline base/REPL.jl]: any way to test this?

File: base/boot.jl [hunk 2]
```diff
@@ -332,6 +333,39 @@ end
...
+print(io::IO, x::String) = write(io, x)
+print(io::IO, x::ANY) = show(io, x)
+print(io::IO, a::ANY...) = for x in a; print(io, x); end
+println(io::IO) = write(io, 0x0a) # 0x0a = '\n'
+println(io::IO, x::ANY...) = (print(io, x...); println(io))
+
+println() = println(STDOUT)
+println(a::ANY) = println(STDOUT, a)
+print(a::ANY) = print(STDOUT, a)
+show(a::ANY) = show(STDOUT, a)
```
Comment (tkelman) [inline base/boot.jl]: should these 3 be splatting?

File: base/coreimg.jl [hunk 3]
```diff
@@ -2,20 +2,24 @@
...
+import Core: print, println, show, write, unsafe_write, STDOUT, STDERR
+if false # show that the IO system is already (relatively) operational
+    print("HELLO")
+    println(" WORLD")
+    show("αβγ :)"); println()
+    println(STDERR, "TEST")
+    println(STDERR, STDERR)
+    println(STDERR, 'a')
+    println(STDERR, 'α')
+end
```
Comment (tkelman) [inline base/coreimg.jl]: Should this be commented out or deleted or something? I'm guessing we don't have any core-only tests?

File: base/sysimg.jl [hunk 4]
```diff
@@ -19,11 +19,21 @@ include("exports.jl")
...
-    print(a::ANY...) = for x=a; print(x); end
+    # otherwise, they just just eventually get (noisily) overwritten later
+    global show, print, println
+    show(io::IO, x::ANY) = Core.show(io, x)
+    print(io::IO, a::ANY...) = Core.print(io, a...)
+    println(io::IO, x::ANY...) = Core.println(io, x...)
+    if false # show that the IO system now (relatively) operational
+        print("HELLO")
+        println(" WORLD")
+        show("αβγ :)"); println()
```
Comment (tkelman) [inline base/sysimg.jl]: I feel compelled to suggest the `α` be replaced with `γδ` if this isn't removed.

File: src/jl_uv.c [hunk 5]
```diff
@@ -427,10 +427,7 @@ JL_DLLEXPORT void jl_uv_writecb(uv_write_t *req, int status)
     }
 }
 
-// Note: jl_write() is called only by jl_vprintf().
-// See: doc/devdocs/stdio.rst
```
Comment (tkelman) [inline src/jl_uv.c]: so do these need updating?

File: base/boot.jl (hunk 2 above)
Comment (vtjnash) [inline base/boot.jl]: yes, fixed (also fixed the function, which previously required features not available to Core)

File: src/jl_uv.c (hunk 5 above)
Comment (vtjnash) [inline src/jl_uv.c]: it's really odd that we documented a `static` function, but apparently we did. Updated that with the new name

File: base/coreimg.jl (hunk 3 above)
Comment (vtjnash) [inline base/coreimg.jl]: It's disabled. I also added tests, although those don't run on quite as minimal an environment

Comment (tkelman) [inline base/coreimg.jl]: seems really odd to leave in dead testing code here and in `sysimg.jl`

**Pull request merged**